### PR TITLE
Feature/fga/push subscribe to room

### PR DESCRIPTION
**Pull request by ganfra**
## Content
Add a feature flag so we can try subscribing to room whenever we receive a notifiable push.

## Motivation and context
To improve time to see timeline when opening the app from push notification.

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/OnNotifiableEventReceived.kt [hunk 1]
```diff
@@ -32,10 +35,23 @@ interface OnNotifiableEventReceived {
...
         }
     }
+
+    private fun CoroutineScope.subscribeToRoomIfNeeded(notifiableEvent: NotifiableEvent) = launch {
+        if (!featureFlagService.isFeatureEnabled(FeatureFlags.SyncOnPush)) {
+            return@launch
+        }
+        val client = matrixClientProvider.getOrRestore(notifiableEvent.sessionId).getOrNull() ?: return@launch
+        client.getRoom(notifiableEvent.roomId)?.use { room ->
+            room.subscribeToSync()
```
Comment (jmartinesp) [inline libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/OnNotifiableEventReceived.kt:54]: I think that might be instantiating a timeline too?

**Comment by jmartinesp** [comment]
After enabling the developer option I see no `/sync` in the logs when receiving a notification. So maybe we need to start the sync service until we receive a first sync?

File: libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/OnNotifiableEventReceived.kt (hunk 1 above)
Comment (ganfra) [reply libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/OnNotifiableEventReceived.kt:54]: Yes the goal is also to create and cache a timeline before clicking the notification

**Comment by jmartinesp** [comment]
Actually my bad, I thought this PR was about opportunistic syncs when push notifications were received, so we could get updates for other subscribed rooms too, but it's only about optimising loading times for the timeline, right? 

To be honest, I see no difference with or without the developer option in the debug build. Maybe it'll be more noticeable in the nightlies. I'll approve it so we can test it there.

**Review by jmartinesp — APPROVED** [review]

**Comment by ganfra** [comment]
Ok, lets see and remove it if it's useless

**Pull request merged**